pkg/host/internal/sriov/sriov.go

```diff
@@ -467,76 +502,184 @@ func (s *sriov) ConfigSriovInterfaces(storeManager store.ManagerInterface,
+			log.Log.Error(err, "ConfigSriovInterfaces(): fail to configure sriov interface. resetting interface.", "address", iface.iface.PciAddress)
+			if iface.iface.ExternallyManaged {
+				log.Log.Info("ConfigSriovInterfaces(): skipping device reset as the nic is marked as externally created")
 			} else {
-				err = s.udevHelper.RemoveUdevRule(ifaceStatus.PciAddress)
-				if err != nil {
-					return err
+				if resetErr := s.ResetSriovDevice(iface.ifaceStatus); resetErr != nil {
+					log.Log.Error(resetErr, "ConfigSriovInterfaces(): failed to reset on error SR-IOV interface")
 				}
 			}
+			return err
+		}
 
-			if err = s.ResetSriovDevice(ifaceStatus); err != nil {
-				return err
-			}
+		// Save the PF status to the host
+		err := storeManager.SaveLastPfAppliedStatus(&iface.iface)
+		if err != nil {
+			log.Log.Error(err, "ConfigSriovInterfaces(): failed to save PF applied config to host")
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *sriov) resetSriovInterfaces(storeManager store.ManagerInterface, interfaces []sriovnetworkv1.InterfaceExt) error {
+	for _, iface := range interfaces {
+		if err := s.checkForConfigAndReset(iface, storeManager); err != nil {
+			log.Log.Error(err, "resetSriovInterfaces(): fail to configure sriov interface. resetting interface.", "address", iface.PciAddress)
+			return err
+		}
+	}
+	log.Log.V(2).Info("resetSriovInterfaces(): sriov reset finished")
+	return nil
+}
+
+func checkNeedUpdate(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetworkv1.InterfaceExt, storeManager store.ManagerInterface) (bool, error) {
+	if !sriovnetworkv1.NeedToUpdateSriov(iface, ifaceStatus) {
+		log.Log.V(2).Info("ConfigSriovInterfaces(): no need update interface", "address", iface.PciAddress)
+
+		// Save the PF status to the host
+		err := storeManager.SaveLastPfAppliedStatus(iface)
+		if err != nil {
+			log.Log.Error(err, "ConfigSriovInterfaces(): failed to save PF applied config to host")
+			return false, err
 		}
+
+		return true, nil
 	}
+	return false, nil
+}
+
+func (s *sriov) checkForConfigAndReset(ifaceStatus sriovnetworkv1.InterfaceExt, storeManager store.ManagerInterface) error {
+	// load the PF info
+	pfStatus, exist, err := storeManager.LoadPfsStatus(ifaceStatus.PciAddress)
+	if err != nil {
+		log.Log.Error(err, "checkForConfigAndReset(): failed to load info about PF status for device",
+			"address", ifaceStatus.PciAddress)
+		return err
+	}
+
+	if !exist {
+		log.Log.Info("checkForConfigAndReset(): PF name with pci address has VFs configured but they weren't created by the sriov operator. Skipping the device reset",
+			"pf-name", ifaceStatus.Name,
+			"address", ifaceStatus.PciAddress)
+		return nil
+	}
+
+	if pfStatus.ExternallyManaged {
+		log.Log.Info("checkForConfigAndReset(): PF name with pci address was externally created skipping the device reset",
+			"pf-name", ifaceStatus.Name,
+			"address", ifaceStatus.PciAddress)
+		return nil
+	} else {
```

Comment:
true branch returns, you can simplify by removing else block and moving the removal of udev rule to the containing block.